static/js/message_util.js

```diff
@@ -20,8 +20,10 @@ function add_messages(messages, msg_list, opts) {
 exports.add_old_messages = function (messages, msg_list) {
     return add_messages(messages, msg_list, {messages_are_new: false});
 };
-exports.add_new_messages = function (messages, msg_list) {
-    return add_messages(messages, msg_list, {messages_are_new: true});
+
+exports.add_new_messages = function (messages, msg_list, opts = {}) {
+    opts.messages_are_new = true;
+    return add_messages(messages, msg_list, opts);
 };
```

Comment:
So, following the changes suggested above, I think this function could read:
```
if (!msg_list.data.has_found_newest()) {
    // Comment explaining this corner case
    message_list.data.fetch_status.update_expected_max_message_id(messages)
   return;
}
return add_messages(messages, msg_list, {messages_are_new: true});
```

And then the `FetchStatus` logic would need to do the loop over message objects, but that's easy enough.